test/models/comment_test.rb

```diff
@@ -122,15 +124,19 @@ def do_comment_test(chg, obs, user, summary, comment = "")
   end
 
   def num_emails
-    ActionMailer::Base.deliveries.length
+    # ActionMailer::Base.deliveries.length
+    QueuedEmail.count
   end
 
+  # FIXME : need to alter this to test for QueuedEmails
   def sent_emails(start)
     return "No emails were sent" if num_emails == start
 
-    strs = ActionMailer::Base.deliveries[start..-1].map do |mail|
-      "to: #{mail["to"]}, subject: #{mail["subject"]}"
-    end
-    "These emails were sent:\n#{strs.join("\n")}"
+    all_emails = QueuedEmail.all[start..-1]
+    binding.break
+    # strs = ActionMailer::Base.deliveries[start..-1].map do |mail|
+    #   "to: #{mail["to"]}, subject: #{mail["subject"]}"
+    # end
+    # "These emails were sent:\n#{strs.join("\n")}"
   end
```

Comment:
After converting Comment emails to use QE, this sub-test (used in many comment tests) fails. I'm going to play with this and see if i can get it to check the recently queued emails for the `to` and the `subject`

Reply:
Same issue. The only QE returned by `all_emails` is
```ruby
[#<QueuedEmail::Webmaster:0x00007fed92f92bf8
  id: 3379,
  user_id: nil,
  queued: Tue, 15 Aug 2023 18:14:31.000000000 EDT -04:00,
  num_attempts: nil,
  flavor: "QueuedEmail::Webmaster",
  to_user_id: nil>]
```
It seems like the last one should be a `QueuedEmail::CommentAdd`. 
Do queued email records not get added immediately, or something?

Reply:
Aha - I needed to set 
```ruby
  def setup
    QueuedEmail.queue = true
  end
```
Do i also need to set it to `false` during `teardown`?

Reply:
Set to false during teardown, yes.